src/runtimes/node/bundlers/nft/index.ts

```diff
@@ -81,10 +72,12 @@ const traceFilesAndTranspile = async function ({
   pluginsModulesPath?: string
 }) {
   const fsCache: FsCache = {}
-  const cache: NftCache = {}
-  const { fileList: dependencyPaths } = await nodeFileTrace([mainFile], {
+  const {
+    fileList: dependencyPaths,
+    esmFileList,
```

Comment:
We were previously relying on NFT's cache object to inspect which of the paths corresponded to ESM files, but I realised there's a `esmFileList` property that contains just that, so there's no need to muck about with the internal cache.